src/MUIDataTable.js

```diff
@@ -831,6 +831,7 @@ class MUIDataTable extends React.Component {
     const cleanRows = data.filter(({ index }) => !selectedMap[index]);
 
     if (this.options.onRowsDelete) {
+      if (this.options.onRowsDelete(selectedRows) === false) return;
       this.options.onRowsDelete(selectedRows);
```

Comment:
This line should have been removed. Now when onRowsDelete doesn't return false, it is called again immediately.

Reply:
You are correct, nice catch! I will push a fix.

Reply:
https://github.com/gregnb/mui-datatables/pull/631